spec/controllers/api/enterprise_fees_controller_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+
+module Api
+  describe EnterpriseFeesController, type: :controller do
+    include AuthenticationWorkflow
+
+    let(:unreferenced_fee) { create(:enterprise_fee) }
+    let(:referenced_fee) { create(:enterprise_fee) }
+    let(:product) { create(:product) }
+    let(:distributor) { create(:distributor_enterprise) }
+    let!(:product_distribution) { create(:product_distribution, product: product, distributor: distributor, enterprise_fee: referenced_fee) }
+    let(:current_user) { create(:admin_user) }
+
+    before do
+      allow(controller).to receive(:spree_current_user) { current_user }
+    end
+
+    describe "destroy" do
+      it "removes the fee" do
+        spree_delete :destroy, id: unreferenced_fee.id, format: :json
+        expect(response).to be_success
+      end
+
+      context "when the fee is referenced by a product distribution" do
+        it "should render 403" do
+          spree_delete :destroy, id: referenced_fee.id, format: :json
+          expect(response.status).to eq 403
+        end
+
+        it "should display the correct error message" do
+          spree_delete :destroy, id: referenced_fee.id, format: :json
+          expect(response.body).to match(/That enterprise fee cannot be deleted/)
+        end
```

Comment:
I think I would one more detail to these tests, here we are just checking the response code, but maybe we should check what has happened to the data as well. So under `it "removes the fee"` we could check the fee has actually been deleted from the database (as well as the response), and in the other we can check that it has not been deleted. 

I'd maybe merge the two tests for the referenced fee into one test, like `it "does not remove the fee"` and just have a couple of `expect` lines in the single test to check it's not deleted and check the response/message.